#Exercise 4 Using CensusReporter to Make Geographic Selections.

In [0]:
import pandas as pd
import requests 
import json
from google.colab import drive
drive.mount('/content/drive')
df = pd.read_csv(open('drive/My Drive/Colab Notebooks/msas.csv', 'rb'))

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [0]:
df.head()

CBSA                         NAME  POPESTIMATE2017
0  10180                  Abilene, TX           170219
1  10420                    Akron, OH           703505
2  10500                   Albany, GA           151434
3  10540                   Albany, OR           125047
4  10580  Albany-Schenectady-Troy, NY           886188


Should we look for areas with large Hispanic populations parameter 3. Hispanic

---B03002 Hispanic 

In [0]:
geoid = '31000US' + df['CBSA'].astype(str)
half1 = geoid[:len(geoid)//2]
half2 = geoid[len(geoid)//2:]
half1 = ",".join(half1)
half2 = ",".join(half2)

#Question1
##1 What city has the highest count of the audience you chose for parameter #1?

In [0]:
t1 = ['B19001']
t01 = ",".join(t1)

In [0]:
requesturl = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (t01, half1)
loadedjson = requests.get(requesturl)
parsedjson = json.loads(loadedjson.text)


In [0]:
requestur2 = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (t01, half2)
loadedjson2 = requests.get(requestur2)
parsedjson2 = json.loads(loadedjson2.text)

What average HH income should we choose? parameter 1

---B19001 Household income 50k to 74.9k

In [0]:
df1 = pd.DataFrame(columns=["country", "population"], index=list(range(1,len(geoid))))

In [0]:
#first part
for i in list(range(1,len(geoid)//2)):
  df1['country'][i]=[geoid[i]]
  df1["population"][i]=parsedjson['data'][geoid[i]]['B19001']['estimate']['B19001011']+parsedjson['data'][geoid[i]]['B19001']['estimate']['B19001012']

In [0]:
#second part
for i in list(range(192,len(geoid))):
  df1['country'][i]=[geoid[i]]
  df1["population"][i]=parsedjson2['data'][geoid[i]]['B19001']['estimate']['B19001011']+parsedjson2['data'][geoid[i]]['B19001']['estimate']['B19001012']

In [0]:
print(df1.loc[df1["population"] == max(df1["population"])])

            country   population
249  [31000US35620]  1.01121e+06


In [0]:
print(df.loc[df['CBSA']==35620])

      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


Based the calculation New York_Newark-Jersey City has the highest count that range between 50k to 74.9k is 1.01121e+06

#Question2
##2 What city has the highest count of the audience you chose for parameter #2?

In [0]:
t2 = ['B01001']
t02 = ",".join(t2)

In [0]:
requestur3 = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (t02, half1)
loadedjson3 = requests.get(requestur3)
parsedjson3 = json.loads(loadedjson3.text)
requestur4 = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (t02, half2)
loadedjson4 = requests.get(requestur4)
parsedjson4 = json.loads(loadedjson4.text)

In [0]:
#age range from 18 to 34
ageran = ['B01001007','B01001008','B01001009','B01001010','B01001011', 'B01001012']

In [0]:
df2 = pd.DataFrame(columns=["location", "population"], index=list(range(1,len(geoid))))

In [0]:
for i in list(range(1,len(geoid)//2)):
  df2['location'][i]=[geoid[i]]
  df2["population"][i]=parsedjson3['data'][geoid[i]]['B01001']['estimate'][ageran[0]]+ \
    parsedjson3['data'][geoid[i]]['B01001']['estimate'][ageran[1]]+\
    parsedjson3['data'][geoid[i]]['B01001']['estimate'][ageran[2]]+\
    parsedjson3['data'][geoid[i]]['B01001']['estimate'][ageran[3]]+\
    parsedjson3['data'][geoid[i]]['B01001']['estimate'][ageran[4]]+\
    parsedjson3['data'][geoid[i]]['B01001']['estimate'][ageran[5]]



In [0]:
for i in list(range(192,len(geoid))):
  df2['location'][i]=[geoid[i]]
  df2["population"][i]=parsedjson4['data'][geoid[i]]['B01001']['estimate'][ageran[0]]+ \
    parsedjson4['data'][geoid[i]]['B01001']['estimate'][ageran[1]]+\
    parsedjson4['data'][geoid[i]]['B01001']['estimate'][ageran[2]]+\
    parsedjson4['data'][geoid[i]]['B01001']['estimate'][ageran[3]]+\
    parsedjson4['data'][geoid[i]]['B01001']['estimate'][ageran[4]]+\
    parsedjson4['data'][geoid[i]]['B01001']['estimate'][ageran[5]]

In [0]:
print(df2.loc[df2["population"] == max(df2["population"])])
print(df.loc[df['CBSA'] == 35620])

           location   population
249  [31000US35620]  2.33054e+06
      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


As we can see the New York-Newark-Jersey City has the highest count and age between 18 to 34 is 2.33054e+06

#Question3
##3 What city has the highest count of the audience you chose for parameter #3?

In [0]:
t3 = ['B03003']
t03 = ",".join(t3)

In [0]:
requestur5 = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (t03, half1)
loadedjson5 = requests.get(requestur5)
parsedjson5 = json.loads(loadedjson5.text)
requestur6 = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (t03, half2)
loadedjson6 = requests.get(requestur6)
parsedjson6 = json.loads(loadedjson6.text)

In [0]:
df3 = pd.DataFrame(columns=["location", "population"], index=list(range(1,len(geoid))))

In [0]:
for i in list(range(1,len(geoid)//2)):
  df3['location'][i]=[geoid[i]]
  df3["population"][i]=parsedjson5['data'][geoid[i]]['B03003']['estimate']['B03003003']

In [0]:
for i in list(range(192,len(geoid))):
    df3['location'][i]=[geoid[i]]
    df3["population"][i]=parsedjson6['data'][geoid[i]]['B03003']['estimate']['B03003003']

In [0]:
print(df3.loc[df3["population"] == max(df3["population"])])
print(df.loc[df['CBSA'] == 31080])

           location   population
210  [31000US31080]  6.00354e+06
      CBSA                                NAME  POPESTIMATE2017
210  31080  Los Angeles-Long Beach-Anaheim, CA         13353907


So for the calculation above Los Angeles-Long Beach-Anaheim, CA has the highest Hispanic count which is 6.00354e+06



#Question4
##4 What city has the highest mean percentage across all three categories?

In [0]:
IncomePer = []
agePer = []
HisPer = []
for i, row in df1.iterrows():
  PI = df1['population'][i]/max(df1['population'])
  PA = df2['population'][i]/max(df2['population'])
  PH = df3['population'][i]/max(df3['population'])
  IncomePer.append(PI)
  agePer.append(PA)
  HisPer.append(PH)

In [0]:
df1['Per']=IncomePer
df2['Per']=agePer
df3['Per']=HisPer

In [0]:
#mean percentage for thoes three
mea = []
for i, row in df1.iterrows():
  mea.append((df1['Per'][i]+df2['Per'][i]+df3['Per'][i])/3)
df1['percent mean']=mea


In [0]:
print(df1.loc[df1["percent mean"] == max(df1["percent mean"])])
print(df.loc[df['CBSA'] == 35620])

            country   population       Per  percent mean
249  [31000US35620]  1.01121e+06  0.899115      0.906454
      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


New York-Newark-Jersey City, NY-NJ-PA has the highest percentage mean during those three categories which is 0.906454